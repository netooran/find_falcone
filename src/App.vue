<template>
  <div id="app">
    <HelloWorld :msg="appName"/>
    <PlanetPicker v-if="planets && planets.length"/>
    <VehiclePicker v-if="vehicles && vehicles.length && selectedPlanet >= 0"/>
    <i v-if="errors && errors.length" class="fab fa-grav">{{ errorMsg }}</i>
  </div>
</template>

<script>
import { mapState } from "vuex";
import { mapActions } from "vuex";

import HelloWorld from "./components/HelloWorld.vue";
import PlanetPicker from "./components/PlanetPicker.vue";
import VehiclePicker from "./components/VehiclePicker.vue";

export default {
  name: "app",
  components: {
    HelloWorld,
    PlanetPicker,
    VehiclePicker
  },
  computed: mapState(["planets", "vehicles", "selectedPlanet", "errors"]),
  methods: mapActions(["getPlanets", "getVehicles"]),

  data() {
    return {
      appName: "Finding Falcone!",
      errorMsg: " Unfortunately our ravens did not come back. Try again later."
    };
  },

  created() {
    this.getPlanets();
    this.getVehicles();
  }
};
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
