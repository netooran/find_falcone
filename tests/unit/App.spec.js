import Vue from "vue";
import Vuex from "vuex";
import { shallowMount } from "@vue/test-utils";
import App from "@/App.vue";

Vue.use(Vuex);

describe("App", () => {
  let wrapper, planets, vehicles;
  let store, state, actions;

  beforeEach(() => {
    planets = [
      { name: "Toliman", distance: 437 },
      { name: "Proxima Centauri", distance: 1 }
    ];

    vehicles = [
      { name: "Toliman", distance: 437 },
      { name: "Proxima Centauri", distance: 1 }
    ];

    state = { planets: [], vehicles: [], selectedPlanet: undefined };
    actions = { getPlanets: jest.fn(), getVehicles: jest.fn() };

    store = new Vuex.Store({ state, actions });
    wrapper = shallowMount(App, { store });
  });

  it("has the expected html structure when no data given", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has the expected html structure when planet data given", () => {
    store.replaceState({ planets });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("has the expected html structure when vehicles data given", () => {
    store.replaceState({ vehicles });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("has the expected html structure when vehicles data given and a planet is selected", () => {
    store.replaceState({ vehicles, selectedPlanet: 1 });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("has the expected html structure when all data given", () => {
    store.replaceState({ vehicles, selectedPlanet: 1 });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("has the expected html structure on error", () => {
    store.replaceState({ errors: "Some error" });

    expect(wrapper.text()).toMatch(
      "Unfortunately our ravens did not come back. Try again later."
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it("fetches planets and vehicles on creation", () => {
    expect(actions.getPlanets).toHaveBeenCalled();
    expect(actions.getVehicles).toHaveBeenCalled();
  });
});
