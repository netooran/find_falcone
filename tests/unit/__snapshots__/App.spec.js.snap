// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has the expected html structure on error 1`] = `
<div
  id="app"
>
  <helloworld-stub
    msg="Finding Falcone!"
  />
   
  <!---->
   
  <!---->
   
  <i
    class="fab fa-grav"
  >
     Unfortunately our ravens did not come back. Try again later.
  </i>
</div>
`;

exports[`App has the expected html structure when all data given 1`] = `
<div
  id="app"
>
  <helloworld-stub
    msg="Finding Falcone!"
  />
   
  <!---->
   
  <vehiclepicker-stub />
   
  <!---->
</div>
`;

exports[`App has the expected html structure when no data given 1`] = `
<div
  id="app"
>
  <helloworld-stub
    msg="Finding Falcone!"
  />
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;

exports[`App has the expected html structure when planet data given 1`] = `
<div
  id="app"
>
  <helloworld-stub
    msg="Finding Falcone!"
  />
   
  <planetpicker-stub />
   
  <!---->
   
  <!---->
</div>
`;

exports[`App has the expected html structure when vehicles data given 1`] = `
<div
  id="app"
>
  <helloworld-stub
    msg="Finding Falcone!"
  />
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;

exports[`App has the expected html structure when vehicles data given and a planet is selected 1`] = `
<div
  id="app"
>
  <helloworld-stub
    msg="Finding Falcone!"
  />
   
  <!---->
   
  <vehiclepicker-stub />
   
  <!---->
</div>
`;
